src/Linters/NoPHPArrayLiteralsLinter.hack

```diff
@@ -0,0 +1,159 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\Linters;
+
+use type Facebook\HHAST\{
+  ArrayCreationExpression,
+  ArrayIntrinsicExpression,
+  ArrayToken,
+  DictToken,
+  EditableList,
+  EditableNode,
+  ElementInitializer,
+  LeftBracketToken,
+  LeftParenToken,
+  ListItem,
+  RightBracketToken,
+  RightParenToken,
+  VecToken,
+};
+use namespace HH\Lib\{C, Vec};
+use function Facebook\HHAST\Missing;
+
+final class NoPHPArrayLiteralsLinter extends AutoFixingASTLinter<EditableNode> {
+  <<__Override>>
+  protected static function getTargetType(): classname<EditableNode> {
+    return EditableNode::class;
+  }
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    EditableNode $expr,
+    vec<EditableNode> $_parents,
+  ): ?ASTLintError<EditableNode> {
+    if ($expr is ArrayCreationExpression || $expr is ArrayIntrinsicExpression) {
+      return new ASTLintError(
+        $this,
+        'Do not use PHP array literals, use Hack arrays instead.',
+        $expr,
+      );
+    }
+
+    return null;
+  }
+
+  <<__Override>>
+  public function getFixedNode(EditableNode $expr): EditableNode {
+    /**
```

Comment:
`/**` just for doc API documentation comments- doesn't make sense inside a function body; just `/*`